probe/overlay/weave.go

```diff
@@ -162,43 +336,120 @@ func (w *Weave) Report() (report.Report, error) {
 	defer w.mtx.RUnlock()
 
 	r := report.MakeReport()
-	r.Container = r.Container.WithMetadataTemplates(report.MetadataTemplates{
-		WeaveMACAddress:  {ID: WeaveMACAddress, Label: "Weave MAC", From: report.FromLatest, Priority: 17},
-		WeaveDNSHostname: {ID: WeaveDNSHostname, Label: "Weave DNS Name", From: report.FromLatest, Priority: 18},
-	})
-	r.Overlay = r.Overlay.WithMetadataTemplates(report.MetadataTemplates{
-		WeavePeerName: {ID: WeavePeerName, Label: "Peer Name", From: report.FromLatest, Truncate: 17, Priority: 1},
-		WeaveVersion:  {ID: WeaveVersion, Label: "Version", From: report.FromLatest, Priority: 2},
-	})
+	r.Container = r.Container.WithMetadataTemplates(containerMetadata)
+	r.Overlay = r.Overlay.WithMetadataTemplates(weaveMetadata).WithTableTemplates(weaveTableTemplates)
 
 	// We report nodes for all peers (not just the current node) to highlight peers not monitored by Scope
 	// (i.e. without a running probe)
 	// Note: this will cause redundant information (n^2) if all peers have a running probe
 	for _, peer := range w.statusCache.Router.Peers {
-		node := report.MakeNodeWith(report.MakeOverlayNodeID(report.WeaveOverlayPeerPrefix, peer.Name),
-			map[string]string{
-				WeavePeerName:     peer.Name,
-				WeavePeerNickName: peer.NickName,
-			})
-		// Peer corresponding to current host
-		if peer.Name == w.statusCache.Router.Name {
-			node = node.WithLatest(report.HostNodeID, mtime.Now(), w.hostID)
-			node = node.WithLatest(WeaveVersion, mtime.Now(), w.statusCache.Version)
-			node = node.WithParents(report.EmptySets.Add(report.Host, report.MakeStringSet(w.hostID)))
-		}
-		for _, conn := range peer.Connections {
-			if conn.Outbound {
-				node = node.WithAdjacent(report.MakeOverlayNodeID(report.WeaveOverlayPeerPrefix, conn.Name))
-			}
-		}
+		node := w.getPeerNode(peer)
 		r.Overlay.AddNode(node)
 	}
-	if w.statusCache.IPAM.DefaultSubnet != "" {
+	if w.statusCache.IPAM != nil {
 		r.Overlay.AddNode(
-			report.MakeNode(report.MakeOverlayNodeID(report.WeaveOverlayPeerPrefix, w.statusCache.Router.Name)).WithSets(
-				report.MakeSets().Add(host.LocalNetworks, report.MakeStringSet(w.statusCache.IPAM.DefaultSubnet)),
-			),
+			report.MakeNode(report.MakeOverlayNodeID(report.WeaveOverlayPeerPrefix, w.statusCache.Router.Name)).
+				WithSet(host.LocalNetworks, report.MakeStringSet(w.statusCache.IPAM.DefaultSubnet)),
 		)
 	}
 	return r, nil
 }
+
+func (w *Weave) getPeerNode(peer weave.Peer) report.Node {
+	node := report.MakeNode(report.MakeOverlayNodeID(report.WeaveOverlayPeerPrefix, peer.Name))
+	latests := map[string]string{
+		WeavePeerName:     peer.Name,
+		WeavePeerNickName: peer.NickName,
+	}
+	// Peer corresponding to current host
+	if peer.Name == w.statusCache.Router.Name {
+		latests[report.HostNodeID] = w.hostID
```

Comment:
I wonder if almost everything inside this conditional should be moved to yet another function? It's purpose isn't super clear to me, so giving it a name might help. Also, it would make the flow of `getPeerNode` more obvious to those who don't know what it's for.